src/main/java/graphql/nadel/result/ResultComplexityAggregator.java

```diff
@@ -0,0 +1,46 @@
+package graphql.nadel.result;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ResultComplexityAggregator {
+    private Integer totalNodeCount;
+    private LinkedHashMap<String, Integer> serviceNodeCountsMap;
+
+
+    public ResultComplexityAggregator(Integer totalNodeCount, LinkedHashMap<String, Integer> serviceNodeCountsMap) {
```

Comment:
You should not take starting state here - up above you require the caller to know to pass in 0 and initial map

Dont do this.

Have the state of the object inside the class - its its business


Also dont use specific types.   
`private LinkedHashMap<String, Integer> serviceNodeCountsMap;`

Always use generic types where you can - and thats almost everywhere